GCRCatalogs/register.py

```diff
@@ -77,6 +79,63 @@ def load_catalog_from_config_dict(catalog_config):
     )(**catalog_config)
 
 
+def write_to_user_config(d, filename=_USER_CONFIG_NAME, overwrite=True):
+    """
+    Write key/value pairs in dict d to the user's config file. 
+
+    Parameters
+    ----------
+    filename : the config file to write to.  Create if necessary
+    overwrite : whether to overwrite old values in an existing file
+
+    Returns
+    -------
+    dict written to the file.  
+    If overwrite condition is violated, return None
+    
+    """
+    config_dir = get_config_dir(create=True)
+    config_path = os.path.join(config_dir, filename)
+    config_dict = {}
+
+    if os.path.exists(config_path):
+        config_dict = load_yaml_local(config_path)
+        if not overwrite:
+            if not config_dict.keys().isdisjoint(d.keys()):
+                warnings.warn("Overwrite condition violated; config file not updated")
+                return None
+
+    for k,v in d.items():
+        config_dict[k] = v
```

Comment:
```suggestion
    config_dict = dict(d)
```

Reply:
config_dict might already have contents if there was an old file.  In that case I want to add the values from d, not discard old contents of config_dict

Reply:
Ah, in that case, use `config_dict.update(d)`, which would make your intend very clear.